R/write.R

```diff
@@ -187,26 +187,31 @@ format_tsv <- function(x, na = "NA", append = FALSE, col_names = !append, quote_
 #' # converted to ISO8601.
 #' x <- parse_datetime("2016-01-01")
 #' str(output_column(x))
-output_column <- function(x) {
+output_column <- function(x, name) {
   UseMethod("output_column")
 }
 
 #' @export
-output_column.default <- function(x) {
+output_column.default <- function(x, name) {
   if (!is.object(x)) return(x)
   as.character(x)
 }
 
 #' @export
-output_column.double <- function(x) {
+output_column.double <- function(x, name) {
   x
 }
 
 #' @export
-output_column.POSIXt <- function(x) {
+output_column.POSIXt <- function(x, name) {
   format(x, "%Y-%m-%dT%H:%M:%OSZ", tz = "UTC", justify = "none")
 }
 
+#' @export
+output_column.list <- function(x, name) {
+  stop(paste0("Flat files can't store the list column '", name, "'"))
```

Comment:
I think this should use `` ` `` rather than `'`, and you need a `call. = FALSE` on the end. Also `stop()` automatically pastes together it's inputs, you can remove the `paste()` if you want.